plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/CubridUser.java

```diff
@@ -16,62 +16,96 @@
 import org.jkiss.dbeaver.ext.generic.model.GenericSchema;
+import org.jkiss.dbeaver.ext.generic.model.GenericStructContainer;
 import org.jkiss.dbeaver.ext.generic.model.GenericTable;
+import org.jkiss.dbeaver.ext.generic.model.GenericTableColumn;
+import org.jkiss.dbeaver.ext.generic.model.GenericTableIndex;
+import org.jkiss.dbeaver.ext.generic.model.GenericTableIndexColumn;
 import org.jkiss.dbeaver.ext.generic.model.GenericView;
+import org.jkiss.dbeaver.ext.generic.model.TableCache;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCResultSet;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCSession;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCStatement;
+import org.jkiss.dbeaver.model.impl.jdbc.JDBCConstants;
+import org.jkiss.dbeaver.model.impl.jdbc.JDBCUtils;
+import org.jkiss.dbeaver.model.impl.jdbc.cache.JDBCCompositeCache;
 import org.jkiss.dbeaver.model.meta.Property;
 import org.jkiss.dbeaver.model.runtime.DBRProgressMonitor;
+import org.jkiss.dbeaver.model.struct.rdb.DBSIndexType;
+import org.jkiss.utils.CommonUtils;
 
+import java.sql.DatabaseMetaData;
+import java.sql.SQLException;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Locale;
 
 public class CubridUser extends GenericSchema
 {
-
     private String name;
     private String comment;
+    private final CubridIndexCache cubridIndexCache;
 
-    public CubridUser(GenericDataSource dataSource, String schemaName, String comment)
-    {
+    public CubridUser(
+            @NotNull GenericDataSource dataSource,
+            @NotNull String schemaName,
+            @Nullable String comment) {
         super(dataSource, null, schemaName);
         this.name = schemaName;
         this.comment = comment;
+        this.cubridIndexCache = new CubridIndexCache(this.getTableCache());
     }
 
+    @NotNull
     @Property(viewable = true, order = 1)
-    public String getName()
-    {
+    public String getName() {
         return name;
     }
 
     @Nullable
     @Property(viewable = true, order = 2)
-    public String getComment()
-    {
+    public String getComment() {
         return comment;
     }
 
-    public boolean supportsSystemTable()
-    {
+    @NotNull
+    public boolean supportsSystemTable() {
         return name.equals("DBA");
     }
 
-    public boolean supportsSystemView()
-    {
+    @NotNull
+    public boolean supportsSystemView() {
         return name.equals("DBA");
     }
 
-    public boolean showSystemTableFolder()
-    {
+    @NotNull
+    public boolean showSystemTableFolder() {
         return this.getDataSource().getContainer().getNavigatorSettings().isShowSystemObjects();
     }
 
+    @NotNull
+    public boolean supportsSynonym() {
+        return ((CubridDataSource) this.getDataSource()).getSupportMultiSchema();
+    }
+
+    @NotNull
+    public boolean supportsTrigger() {
+        return this.getDataSource().getContainer().getConnectionConfiguration().getUserName().equals("dba");

```

Comment:
I do not see any restrictions on creating a Cubrid connection with an empty user name. So, please add not null condition here.

Reply:
I have fixed this.